### PR TITLE
Add basic Socket.io setup

### PR DESCRIPTION
**Pull request by bomanimc**
Pair-programmed setup for adding a Node.js server to run our Socket.io setup on the backend.

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -1,6 +1,20 @@
...
+  process.env.NODE_ENV === 'production'
+    ? 'https://clapcitycinema.herokuapp.com'
+    : 'http://localhost:3000';
+const socket = io(socketURL);
 
 export default function Index() {
+  useEffect(() => {
+    socket.on('message', data => {
+      console.log(data);
+      // socket.emit('my other event', { my: 'data' });
```
Comment (bomanimc) [inline pages/index.js:14]: nit: rm

File: server.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    res.json(messages);
+  });
+
+  app.get('*', (req, res) => {
+    return nextHandler(req, res);
+  });
+
+  server.listen(port, err => {
+    if (err) throw err;
+    console.log('> Ready on http://localhost:3000');
```
Comment (bomanimc) [inline server.js:35]: Template literal with pot

File: server.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+const app = require('express')();
+const server = require('http').Server(app);
+const io = require('socket.io')(server);
+const next = require('next');
+
+const dev = process.env.NODE_ENV !== 'production';
+const port = parseInt(process.env.PORT, 10) || 3000;
```
Comment (jbirori) [inline server.js:7]: Don't think you need to use parseInt here, but a good safety measure.

File: server.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    res.json(messages);
+  });
+
+  app.get('*', (req, res) => {
+    return nextHandler(req, res);
+  });
+
+  server.listen(port, err => {
+    if (err) throw err;
+    console.log(`> Ready on http://localhost:${port}`);
```
Comment (jbirori) [inline server.js:35]: Might be useful to make this a bit more dynamic: `> Ready on http://${hostname}:${port}`

File: server.js (hunk 3 above)
Comment (bomanimc) [reply server.js:7]: makes sense. Gonna remove!

File: server.js (hunk 4 above)
Comment (bomanimc) [reply server.js:35]: Learned that it might not be possible to get hostname

**Pull request merged**